### PR TITLE
Show screenshot thumbnail on package page

### PR DESCRIPTION
**Pull request by szakarias**

### CODE REVIEW

**Comment by szakarias** [comment]
image

**Comment by isoos** [comment]
I'm wondering if the thumbnail would be better at the end of the metadata section. With that, we would start the info box with the scores and the publisher (both pub.dev-specific), and then continue with user-provided information, a short description and then the screenshots. wdyt?

File: app/lib/frontend/templates/views/pkg/screenshots.dart [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    children: [prev, imageContainer, next],
+  );
+}
+
+d.Node screenshotThumbnailNode(
+    String thumbnailUrl, List<String>? screenshotUrls) {
+  final collectionsIconWhite =
+      staticUrls.getAssetUrl('/static/img/collections_white_24dp.svg');
+  return d.div(attributes: {
+    'thumbnail-data': screenshotUrls!.reduce((a, b) => '$a,$b'),
```
Comment (isoos) [inline app/lib/frontend/templates/views/pkg/screenshots.dart:46]: maybe `screenshotUrls!.join(',')` ?

File: pkg/web_app/lib/src/screenshot_carousel.dart [hunk 2]
```diff
@@ -57,7 +57,7 @@ void _setEventForScreenshot() {
   for (final thumbnail in thumbnails) {
     thumbnail.onClick.listen((event) {
       showElement(carousel);
-      images = thumbnail.attributes['data']!.split(',');
+      images = thumbnail.attributes['thumbnail-data']!.split(',');
```
Comment (isoos) [inline pkg/web_app/lib/src/screenshot_carousel.dart:60]: I think this should rather be `data-<x>` attribute, that can be accessed via `Element.dataset['x']`.

File: app/lib/frontend/templates/views/pkg/screenshots.dart (hunk 1 above)
Comment (szakarias) [reply app/lib/frontend/templates/views/pkg/screenshots.dart:46]: done

File: pkg/web_app/lib/src/screenshot_carousel.dart (hunk 2 above)
Comment (szakarias) [reply pkg/web_app/lib/src/screenshot_carousel.dart]: Definitely! :-) Done.

**Review by jonasfj — APPROVED** [review]

**Comment by szakarias** [comment]
Landing this as per discussion, it is still behind experimental. We will probably move the placement of the thumbnail once we land on a final design.

**Pull request merged**